### PR TITLE
Improve serde performance for Int96 blocks

### PR DESCRIPTION
**Pull request by skrzypo987**

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]
let's land tests first and then this with tests

File: presto-spi/src/main/java/io/prestosql/spi/block/Int96ArrayBlockEncoding.java [hunk 1]
```diff
@@ -38,11 +40,25 @@ public void writeBlock(BlockEncodingSerde blockEncodingSerde, SliceOutput sliceO
...
+            sliceOutput.writeBytes(getLowSlice(block));
+        }
+        else {
+            long[] high = new long[positionCount];
+            int[] low = new int[positionCount];
+            int nonNullPositionCount = 0;
+            for (int i = 0; i < positionCount; i++) {
+                high[nonNullPositionCount] = block.getLong(i, 0);
+                low[nonNullPositionCount] = block.getInt(i, 8);
+                if (!block.isNull(i)) {
```
Comment (zaz968m) [inline presto-spi/src/main/java/io/prestosql/spi/block/Int96ArrayBlockEncoding.java:54]: Do I remember correctly that this check before previous two lines performs worse?

Comment (skrzypo987) [reply presto-spi/src/main/java/io/prestosql/spi/block/Int96ArrayBlockEncoding.java:54]: Mostly yes. It is a bit faster for datasets with a lot of nulls (>81% to be precise), but other than that it is much slower

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**